src/go/rpk/pkg/config/manager.go

```diff
@@ -342,8 +342,18 @@ func checkAndWrite(fs afero.Fs, v *viper.Viper, path string) error {
 			return err
 		}
 	}
+	current := viper.New()
+	current.SetFs(fs)
+	current.SetConfigFile(path)
+	err = current.ReadInConfig()
+	if err != nil {
+		// Couldn't read the current config.
+		return err
+	}
+	// Merge the new config onto the previous one.
+	current.MergeConfigMap(v.AllSettings())
 	log.Debugf("Writing the new redpanda config to '%s'", path)
-	err = write(v, path)
+	err = write(current, path)
```

Comment:
What's the meaning of the return at line [#328](https://github.com/vectorizedio/redpanda/pull/243/files#diff-b6df917c73b6ea330ec97880f51ec02d486fa2ec21cf3b69455ed82064173f02R328)? It seems to not merge env with default?

It's getting pretty hard to figure out what's going on vs. what's intended.

Reply:
If there isn't a file at the same path, it just writes the config. That is, there's no need to merge anything since there isn't anything.

Reply:
If there was a function, `UpdateConfigFile(fs afero.Fs, configFilePath string) error`, what would it do?

In my mind:

1. Construct a manager
2. Read config
  2.1 Generate default config (in memory)
  2.1 Set or override from config file
  2.2. Set or override from env
  2.3. Set or override from command line
3. Write Config
  3.1 Backup existing file
  3.2 Write config file

What about:
`GetConfig(srcs ...ConfigSource) error`
`SetConfig(sinks ...ConfigSink) error`
`UpdateConfig(sink ConfigSink, srcs ...ConfigSource) error`